src/Utilities/functions.jl

```diff
@@ -121,6 +121,36 @@ function Base.getindex(it::ScalarFunctionIterator{MOI.VectorAffineFunction{T}},
     MOI.VectorAffineFunction(terms, constant)
 end
 
+"""
+    iscanonical(f::AbstractFunction)
+
+Returns a Bool indicating whether the function is in canonical form.
+See [`canonical`](@ref).
+"""
+iscanonical(f::MOI.VectorAffineFunction) = iscanonical(f.terms,
+                                                       t -> (t.output_index, t.scalar_term.variable_index.value),
+                                                       t -> !iszero(t.scalar_term.coefficient))
+iscanonical(f::MOI.ScalarAffineFunction) = iscanonical(f.terms,
+                                                       t -> t.variable_index.value,
+                                                       t -> !iszero(t.coefficient))
```

Comment:
What about `ScalarQuadraticFunction`, `VectorQuadraticFunction`?

Reply:
Can use the new `termindices` and `coefficient` functions to make this generic.

Reply:
I don't know what we'd consider canonical form for them, as it was never supported before.